packages/common/src/utils/conditionalFormatExpressions.ts

```diff
@@ -0,0 +1,156 @@
+ *   - "currency" -> "currency"
+ */
+function stripParameterPrefix(paramName: string): string {
+    const ldPrefix = 'ld.parameters.';
+    const lightdashPrefix = 'lightdash.parameters.';
+
+    if (paramName.startsWith(ldPrefix)) {
+        return paramName.substring(ldPrefix.length);
+    }
+    if (paramName.startsWith(lightdashPrefix)) {
+        return paramName.substring(lightdashPrefix.length);
+    }
+    return paramName;
+}
+
+/**
+ * Removes surrounding quotes from a string if present
+ */
+function removeQuotes(value: string): string {
+    if (
+        (value.startsWith('"') && value.endsWith('"')) ||
+        (value.startsWith("'") && value.endsWith("'"))
+    ) {
+        return value.slice(1, -1);
+    }
+    return value;
+}
+
+/**
+ * Gets the value from either a parameter or a literal (quoted string)
+ */
+function getConditionValue(
+    value: string,
+    parameters: Record<string, unknown>,
+): unknown {
+    // Check if it's a quoted string literal
+    if (
+        (value.startsWith('"') && value.endsWith('"')) ||
+        (value.startsWith("'") && value.endsWith("'"))
+    ) {
+        return value.slice(1, -1);
+    }
+
+    // Otherwise, treat as parameter name - strip ld.parameters. prefix if present
+    const paramName = stripParameterPrefix(value);
+    return parameters[paramName];
+}
+
+/**
+ * Evaluates a condition like: paramName=="value" or paramName!="value"
+ */
+function evaluateCondition(
+    condition: string,
+    parameters: Record<string, unknown>,
+): boolean {
+    // Check for != operator
+    if (condition.includes('!=')) {
+        const [left, right] = condition
+            .split('!=')
+            .map((s: string) => s.trim());
+        const leftValue = getConditionValue(left, parameters);
+        const rightValue = getConditionValue(right, parameters);
+        return leftValue !== rightValue;
+    }
+
+    // Check for == operator
+    if (condition.includes('==')) {
+        const [left, right] = condition
+            .split('==')
+            .map((s: string) => s.trim());
+        const leftValue = getConditionValue(left, parameters);
+        const rightValue = getConditionValue(right, parameters);
+        return leftValue === rightValue;
+    }
+
+    // If no operator, treat as truthy check
+    const value = getConditionValue(condition, parameters);
+    return Boolean(value);
+}
+
+/**
+ * Evaluates a ternary expression: condition ? trueValue : falseValue
+ * Supports == and != operators
+ */
+function evaluateTernaryExpression(
+    expression: string,
+    parameters: Record<string, unknown>,
+): string {
+    // Split by ? to get condition and values
+    const [conditionPart, valuesPart] = expression
+        .split('?')
+        .map((s: string) => s.trim());
+
+    if (!valuesPart) {
+        return expression; // Invalid ternary, return as-is
+    }
+
+    // Split values by : to get trueValue and falseValue
+    const colonIndex = valuesPart.indexOf(':');
```

Comment:
Why not use the same .split method we've been using with the `:`?

Reply:
That is also an alternative valid implementation. 
The main reason why it is used ":" here is because first we want to know if ":" and see if it is a valid ternary expression. (it is easier than checking lenght of the split)